test/test_recipes_integration.py

```diff
@@ -0,0 +1,46 @@
+import pytest
+import os
+import subprocess
+import json
+
+# Each test with recipe and appropriate parameters in one line
+# Using bracket annotation to set it optional (aka xfail)
+TEST_CASES = [
+    "activedata_usage",
+    ["backout_rate"],
+    ["code_coverage --path caps --rev 45715ece25fc"],
+    "code_coverage_by_suite --path caps --rev 45715ece25fc",
+    "config_durations",
+    "files_with_coverage",
+    ["intermittent_tests"],
+    ["intermittent_test_data"],
+    ["raw_coverage --path caps --rev 45715ece25fc"],
+    "test_durations",
+    ["tests_config_times -t test-windows10-64/opt-awsy-e10s"],
+    ["tests_in_duration"],
+    ["try_efficiency"],
+    ["try_usage"],
+    ["try_users"]
+]
+
+
+def load_tests(tests):
+    return [(test[0], True) if isinstance(test, list) else (test, False) for test in tests]
+
+
+@pytest.mark.parametrize("recipe, optional", load_tests(TEST_CASES),
+                         ids=[val[0] for val in load_tests(TEST_CASES)])
+@pytest.mark.skipif(os.getenv("TRAVIS_EVENT_TYPE") != "cron", reason="Not run by cron job")
+def test_recipe_integration(recipe, optional):
+    try:
+        command = "adr --format json {}".format(recipe).strip()
+        data = subprocess.check_output(command, shell=True, stderr=subprocess.STDOUT)
```

Comment:
It's best to avoid `shell=True` with subprocess as it can lead to security vulnerabilities if you're not careful. You can remove that and make sure that the `command` is a list instead. E.g:

    command = ['adr', '--format', 'json']
    command.extend(recipe)

You'll need to convert all the TEST_CASES to a list as well.

Reply:
Thanks I already updated.